### ISSUE TITLE
[Feature Request] Rotated and removed section views

### ISSUE THREAD

**Issue by FEA-eng** [FreeCAD/FreeCAD#8076]
### Is there an existing request for this?

- [X] I have searched the existing requests

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?f=35&t=74105

### Subproject(s) affected?

Techdraw

### Idea description

The new Complex Section tool added support for several advanced section types (#6309). However, rotated and removed sections are not yet supported and have to be created manually with workarounds.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luzpaz** [comment]
Was this recently addressed ?

**Comment by FEA-eng** [comment]
> Was this recently addressed ?

Nope, no development in this regard so far.

**Comment by maxwxyz** [comment]
@FEA-eng not fixed by #31614 right?

**Comment by FEA-eng** [comment]
> [@FEA-eng](https://github.com/FEA-eng) not fixed by [#31614](https://github.com/FreeCAD/FreeCAD/pull/31614) right?

I don't think so; those types of section views are quite specific. But maybe @PaddleStroke could have a look at them too.

Btw., there's also #8075, probably even more important.

**Comment by PaddleStroke** [comment]
I'm not sure to understand what they are compared to regular section view. Can you detail? Or if there a video from another cad.
The feature has a bounty, that's motivating ;)

**Comment by FEA-eng** [comment]
> I'm not sure to understand what they are compared to regular section view. Can you detail?

Here’s a largely shortened definition from the classic Polish book about technical drawings (pictures with examples from it are included in the issue description):

> A rotated/removed section shows the shape of a cross-section rotated 90° and either placed on the object (rotated section) or outside it (removed section). Unlike a regular section, it doesn’t show the outlines of features behind the cutting plane.
> 
> If a hole is cut along its axis, its hidden edges can be shown to keep the section as one piece. Otherwise, a regular section should be used if the rotated/removed section would be split.

> Or if there a video from another cad. The feature has a bounty, that's motivating ;)

There are some examples in the SolidWorks documentation: https://help.solidworks.com/2026/english/solidworks/sldworks/c_drawing_removed_section_view.htm
https://help.solidworks.com/2026/english/SolidWorks/sldworks/c_rotated_section_views.htm?id=13.12.4.7.16

**Comment by PaddleStroke** [comment]
Ok so I think I understand, although I'm not sure.
I found this youtube video of SolidWorks : https://www.youtube.com/watch?v=ab37evb0AGM
Is this implementation good?

Basically if I understand, the only differences with normal section views is that we show ONLY the section plane. Not what is behind. So only the cut planed but not the visible lines that would be visible in a normal section view.

After this, I see that solidworks position the create view automatically centered on the section line and has a dashed line connecting the view to the cutline.

Is this how you guys want it?

It's not clear to me what is the difference between rotated and removed. Is one of them 7) and the other 8) in your screenshot?

**Comment by FEA-eng** [comment]
> Basically if I understand, the only differences with normal section views is that we show ONLY the section plane. Not what is behind. So only the cut planed but not the visible lines that would be visible in a normal section view.

Yes, it's like using the [Part CrossSections](https://wiki.freecad.org/Part_CrossSections) tool. But don't forget the 90 degree rotation.

> After this, I see that solidworks position the create view automatically centered on the section line and has a dashed line connecting the view to the cutline.
> 
> Is this how you guys want it?

For removed yes. See the distinction below.

> It's not clear to me what is the difference between rotated and removed. Is one of them 7) and the other 8) in your screenshot?

**Rotated** is simply placed directly on the view in the spot from which it's taken (and you just rotate it as mentioned above):

<img width="146" height="92" alt="Image" src="https://github.com/user-attachments/assets/51b257f8-96cc-4d63-9d16-e508cd0a70bb" />

Additional examples (from Lewandowski's book):

<img width="404" height="183" alt="Image" src="https://github.com/user-attachments/assets/c930b2db-da8c-469b-bda1-145415ea5aee" />

**Removed** is placed away from the view with those lines connecting it with the view. See other examples in my post such as the shaft (7 and 8 - different ways of positioning):

<img width="417" height="198" alt="Image" src="https://github.com/user-attachments/assets/cae1ce0b-c1bc-4ea4-8a20-26c7d3ab323d" />

Additional examples (from Lewandowski's book):

<img width="424" height="196" alt="Image" src="https://github.com/user-attachments/assets/1a6219da-5467-4ab3-bd1f-308096b45be8" />

Someone shared manually created removed section on the forum [here](https://forum.freecad.org/viewtopic.php?f=35&t=74105#p647763).

Removed seems more common and might be easier to implement, but ideally both types should be available (we can split the issue, though).

**Comment by PaddleStroke** [comment]
But basically it's just the position of the section view. The rotated is just positioned in the middle basically.
So if you are able to click and drag the view and either snap it to the center (as the rotated) or drag it away (like the removed) then is it OK?

**Comment by FEA-eng** [comment]
> So if you are able to click and drag the view and either snap it to the center (as the rotated) or drag it away (like the removed) then is it OK?

Essentially yes, but there might be some additional considerations for rotated sections, such as removing the extension lines and section identifiers and handling overlap (between the section and the view).

**Comment by PaddleStroke** [comment]
Wouldn't a "Show only interesection plane" checkbox to the general section view tool not be enough? Because they are basically the same thing aside from this.
There could also be a "align section view" option to add a cosmetic line and forcing the section view to be aligned.

**Comment by FEA-eng** [comment]
> Wouldn't a "Show only interesection plane" checkbox to the general section view tool not be enough? Because they are basically the same thing aside from this. There could also be a "align section view" option to add a cosmetic line and forcing the section view to be aligned.

Perhaps an extra setting for the existing section view tool would be enough. Possibly, a "Rotated/Removed Section" checkbox that would ensure:
- showing only what's on the cutting plane
- ensuring 90 deg rotation
- allowing placement within or away from the view, with extension lines connecting it to the location of the cut

But another thing is that these sections are taken from arbitrary locations on the view (not necessarily from the origin planes). So there must be a way to define the location of the section. Maybe just what the current Complex Section has.

**Comment by pierreporte** [comment]
In Alibre these sections are created with the normal section tool so it wouldn’t be unheard of to do that in FreeCAD as well. In Alibre you just need to move the latent section view in the right direction for it to appear right: left/right of section plane (when it is vertical) makes the standard section views, along section plane makes rotated/removed section view.

**Comment by franklincg** [comment]
Hi, I'm interested in working on this bounty. Is it still available for a new contributor? I’m currently getting familiar with the TechDraw section-view code and would like to work on it if it’s still open.

**Comment by PaddleStroke** [comment]
Ok I have a good solution for this I think now. It will be in the section rework PR.
